### ISSUE TITLE
datadir to a different location

### ISSUE THREAD

**Issue by stissot** [sous-chefs/mariadb#25]
When the attribute `node['mariadb']['mysqld']['datadir']` is set to a non-default value (e.g `/mnt/mysql`), the server fails to start because the data is not copied from the default location `/var/lib/mysql`
Tested on Ubuntu 14.04 with MariaDB 5.5

**Comment by sinfomicien** [comment]
The question is, must we copy or move the data automatically ? That's why today it's not done... Can't decide what is the best strategy

**Comment by stissot** [comment]
It's difficult to change the datadir in the my.cnf during the package installation. I suggest to keep it to it's default value, but if the `node['mariadb']['mysqld']['datadir']`is different than the default, stop the service, move the data, replace it with a symlink to the new path and restart the service.

**Comment by stissot** [comment]
Hello I've added in a separate branch a shell script that moves the datadir just after install and replaces it by a symlink. I tried to use `service[mysql]` to stop the service but it takes several seconds for the server to be properly shut down and the execute command that moves the directory was notified too quickly so in the end everything is done in the shell script.
A new serverspec test was added that pass on Ubuntu 14.04 with MariaDB 10.0, didn't had time to verify on the other platforms.
